alerts/machine-config-operator/MachineConfigDaemonPivotError.md

```diff
@@ -0,0 +1,146 @@
+You should start troubleshooting
+by examining the main error message and the
+stated reason it gives for not being able to pivot. The following are
+common reasons a pivot can fail.
+
+- The rpm-ostree service is unable to
+pull the image from quay succesfully.
+- There are issues with the rpm-ostree service itself such as
+being unable to start, or unable to build the OsImage folder,
+unable to pivot from the current configuration.
+- The rpm-ostree service gets stuck and client is unable to finish
+the transaction and gets stuck in a loop as a result.
+- The ostree-finalize-staged.service
+  (responsible for rolling out pending updates) is having issues.
+- Networking issues on the node prevent quay from
+being reachable, such as an interface being
+down, a firewall blocking the connection, the proxy, etc.
+
+## Mitigation
+
+- If the machine-config-daemon pod is
+  logging the following errors from rpm-ostree,
+  it is likely that rpm-ostree is stuck in a
+  loop and cannot finish the transaction
+  or there are issues related to the rpm-ostree service itself.
+
+  ```console
+   error: Transaction in progress: (null)
+  ```
+  You should restart the `rpm-ostreed`
+  service on the node that are failing.
+
+  ```console
+  $ oc debug node <Node-name>
+  # chroot /host
+  # systemctl restart rpm-ostreed
+  ```
+- If the machine-config-daemon pod is
+  logging the following errors, it is likely due to networking errors related to
+  timeouts or Internal Server errors in the daemon or controller pods.
+
+     ```console
+    received unexpected HTTP status: 500 Internal Server Error
+     ```
+
+  - Make sure that `quay.io` and
+    it's subdomains are whitelisted by your firewall and proxy.
+    You can test manual pulls with `podman pull`.
+
+    ```console
+    $ podman pull quay.io/startx/couchbase:ubi8 --log-level debug
+    ```
+    - You should also validate that the internal
+     network on the node and the external network
+     are healthy and there are no blockers
+     such as a firewall or switch issues.
+
+- You can also try to force a manual upgrade to the new image
+  if the pivot is stuck.
+
+  ```console
+  -delete the currentconfig(rm /etc/machine-config-daemon/currentconfig)
+  -create a forcefile(touch /run/machine-config-daemon-force) to retry the OS upgrade.
+  ```
+
+  - If you want to dive deeper you can also use the
+    rpm-ostree command line tool to attempt to force a pivot
+
+    ```console
+     rpm-ostree rebase --experimental ostree-unverified-registry:quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+    ```
+
+- If these troubleshooting steps do not fix the problem, gather
+  the following logs, which will be helpful
+  for identifying the issue.
+
+    ```console
+    # rpm-ostree status
+    # journalctl -u ostree-finalize-staged
+    # journalctl -b -1 -u rpm-ostreed.service
+    ```
```

Comment:
```suggestion
- If these troubleshooting steps do not fix the problem, please obtain a must-gather and an sosreport from the affected node. Please attach these files to your support case; this will be helpful for the MCO developer investigating your case.
```
```

Reply:
An sosreport includes these logs and gives us a bigger picture, so let's ask them to do that instead

Reply:
I kind of disagree. I think these runbooks should be written from the perspective that the person doesn't have a subscription to Red Hat or support subs to get external help. So I feel that a runbook should be written telling the person where to find helpful logs. Rather then 'get this dataset and hope you paid  money that someone else will help you.' imo. Let me know if you disagree

Reply:
I think it's better to keep as is for now. As this lets customer access these logs directly on the erroring nodes in the moment.